summertime/model/single_doc/multilingual/base_multilingual_model.py

```diff
@@ -0,0 +1,58 @@
+from summertime.model.single_doc.base_single_doc_model import SingleDocSummModel
+
+import urllib.request
+import fasttext
+
+
+class MultilingualSummModel(SingleDocSummModel):
+
+    lang_tag_dict = None
+
+    def __init__(
+        self,
+        trained_domain: str = None,
+        max_input_length: int = None,
+        max_output_length: int = None,
+    ):
+        super(MultilingualSummModel, self).__init__(
+            trained_domain=trained_domain,
+            max_input_length=max_input_length,
+            max_output_length=max_output_length,
+        )
+
+    @classmethod
+    def assert_summ_input_language(cls, corpus, query):
+        # TODO: add fasttext language detection here
+
+        url = "https://dl.fbaipublicfiles.com/fasttext/supervised-models/lid.176.ftz"
+        # currently using compressed fasttext model
+        urllib.request.urlretrieve(url, "lid.176.ftz")
+        # tqdm(
+        #     urllib.request.urlretrieve(url, "lid.176.ftz"),
+        #     desc="Downloading language detector",
+        # )
+
+        classifier = fasttext.load_model(
+            "./lid.176.ftz"
+        )  # TODO: change download location,
+        # do not redownload every time if not necessary
+
+        if all([isinstance(ins, list) for ins in corpus]):
+            prediction = classifier.predict(corpus[0])
+
+        elif isinstance(corpus, list):
+            prediction = classifier.predict(corpus)
+
+        label = prediction[0][0][0]
```

Comment:
Better to add a comment on the indexing, and possibly a type annotation on the variable prediction if it's a complex compound type.